### PR TITLE
hostname: add support for Anolis distro

### PR DESCRIPTION
**Pull request by geliwei**
Signed-off-by: Liwei Ge <geliwei@openanolis.org>

##### SUMMARY

run following command on Anolis OS

ansible -i config localhost -m hostname -a 'name=geliwei'

ansible report platform not support

##### ISSUE TYPE

- Bugfix Pull Request

##### COMPONENT NAME

modules/hostname
##### ADDITIONAL INFORMATION

```paste below
before 
localhost | FAILED! => {
    "ansible_facts": {
        "discovered_interpreter_python": "/usr/bin/python3.6"
    },
    "changed": false,
    "msg": "hostname module cannot be used on platform Linux (Anolis)"
}

after
localhost | CHANGED => {
    "ansible_facts": {
        "ansible_domain": "",
        "ansible_fqdn": "geliwei",
        "ansible_hostname": "geliwei",
        "ansible_nodename": "geliwei",
        "discovered_interpreter_python": "/usr/bin/python3.6"
    },
    "changed": true,
    "name": "geliwei"
}
```

### CODE REVIEW

File: lib/ansible/modules/hostname.py [hunk 1]
```diff
@@ -803,6 +803,12 @@ class CentOSHostname(Hostname):
     strategy_class = RedHatStrategy
 
 
+class AnolisOSHostname(Hostname):
```
Comment (Akasurde) [inline lib/ansible/modules/hostname.py:806]: @geliwei Thanks for the PR. I will wait for https://github.com/ansible/ansible/pull/74797 to get merged first. Then we can merge this.

Apart from that, please create a [changelog fragment](https://docs.ansible.com/ansible/latest/community/development_process.html#creating-a-changelog-fragment). See [this fragment](https://github.com/ansible/ansible/blob/c2f528b7686c9573aa8493357219e0bbe4416dd3/changelogs/fragments/reboot-change-default-boot-command.yaml) as an example.

**Review by Akasurde — CHANGES_REQUESTED** [review]

File: lib/ansible/modules/hostname.py (hunk 1 above)
Comment (geliwei) [reply lib/ansible/modules/hostname.py:806]: I didn't file a bug for this since it's a minor change, hope that will be OK.

Comment (Akasurde) [reply lib/ansible/modules/hostname.py:806]: Yes that is OK. Just include the PR link in the changelog.

**Review by eqrx — APPROVED** [review]

**Review by Akasurde — APPROVED** [review]
LGTM

**Comment by Akasurde** [comment]
@geliwei Thanks for the contribution. @eqrx Thanks for the review.

**Pull request merged**